primitives/arithmetic/src/fixed.rs

```diff
@@ -0,0 +1,1308 @@
+// Copyright 2020 Parity Technologies (UK) Ltd.
+// This file is part of Substrate.
+
+// Substrate is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Substrate is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Substrate.  If not, see <http://www.gnu.org/licenses/>.
+
+use sp_std::{ops::{self, Add, Sub, Mul, Div}, fmt::Debug, prelude::*, convert::{TryInto, TryFrom}};
+use codec::{Encode, Decode};
+use crate::{
+	helpers_128bit::multiply_by_rational, PerThing,
+	traits::{
+		SaturatedConversion, CheckedSub, CheckedAdd, CheckedMul, CheckedDiv, CheckedNeg,
+		Bounded, Saturating, UniqueSaturatedInto, Zero, One
+	},
+};
+
+#[cfg(feature = "std")]
+use serde::{de, Deserialize, Deserializer, Serialize, Serializer};
+
+pub trait FixedPointOperand: Copy + Clone + Bounded + Zero + Saturating
```

Comment:
pub trait needs documentation + I am very keen on knowing how using any of the below types that implement this trait, will affect the accuracy. for example we have `checked_from_rational(D, D)` and it returns an option and it is directly dependent on this trait. For which data types will it be None?

Reply:
It should return `None` only depending on the `Inner` type, not on the input types. Btw, I'm looking for another name for this trait.

Reply:
Name is okay, just add thorough documentation.